src/encoded/audit/experiment.py

```diff
@@ -61,37 +63,38 @@ def audit_hic_restriction_enzyme_in_libaries(value, system, excluded_types):
         return
     if 'replicates' not in value:
         return
-    frag_methods = set()
-    using_restriction_enzyme = 0
+    
+    
+    fragmentation_methods = set()
+    libraries_fragmentation_methods = {}
+    print ("zopa")
     for rep in value['replicates']:
         rep_lib = rep.get('library')
         rep_status = rep.get('status')
         if rep_lib and rep_status and rep_status not in excluded_types:
             rep_lib_status = rep_lib.get('status')
             if rep_lib_status and rep_lib_status not in excluded_types:
                 if 'fragmentation_method' in rep_lib:
-                    frag_methods.add(rep_lib['fragmentation_method'])
-                    if 'restriction' in rep_lib['fragmentation_method']:
-                        using_restriction_enzyme = 1
+                    libraries_fragmentation_methods[rep_lib.get('@id')] = set(rep_lib.get('fragmentation_method'))
+                    for method in rep_lib.get('fragmentation_method'):
```

Comment:
You can pass list to `.update()` method of a set, instead of iterating through each one.

Reply:
done